File: javascript/test/symmetricEncryptor.test.js
const { SymmetricEncryptor, DecryptionError } = require('../src/symmetricEncryptor');

test('round trip', () => {
  const enc = new SymmetricEncryptor();
  const key = enc.generateKey();
  const ciphertext = enc.encrypt(key, Buffer.from('attack at dawn'));
  expect(enc.decrypt(key, ciphertext).toString()).toBe('attack at dawn');
});

test('rejects tampered ciphertext', () => {
  const enc = new SymmetricEncryptor();
  const key = enc.generateKey();
  const ciphertext = enc.encrypt(key, Buffer.from('secret'));
  ciphertext[ciphertext.length - 1] ^= 0xff;
  expect(() => enc.decrypt(key, ciphertext)).toThrow(DecryptionError);
});

test('nonces differ', () => {
  const enc = new SymmetricEncryptor();
  const key = enc.generateKey();
  const c1 = enc.encrypt(key, Buffer.from('same message'));
  const c2 = enc.encrypt(key, Buffer.from('same message'));
  expect(c1.equals(c2)).toBe(false);
});

File: java/src/test/java/io/github/securekit/SymmetricEncryptorTest.java
package io.github.securekit;

import org.junit.jupiter.api.Test;

import java.nio.charset.StandardCharsets;
import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.*;

class SymmetricEncryptorTest {

    @Test
    void roundTrip() {
        SymmetricEncryptor enc = new SymmetricEncryptor();
        byte[] key = enc.generateKey();
        byte[] plaintext = "attack at dawn".getBytes(StandardCharsets.UTF_8);

        byte[] ciphertext = enc.encrypt(key, plaintext, null);
        byte[] decrypted = enc.decrypt(key, ciphertext, null);
        assertArrayEquals(plaintext, decrypted);
    }

    @Test
    void rejectsTamperedCiphertext() {
        SymmetricEncryptor enc = new SymmetricEncryptor();
        byte[] key = enc.generateKey();
        byte[] ciphertext = enc.encrypt(key, "secret".getBytes(StandardCharsets.UTF_8), null);
        ciphertext[ciphertext.length - 1] ^= 0xFF;

        assertThrows(DecryptionException.class, () -> enc.decrypt(key, ciphertext, null));
    }

    @Test
    void noncesDiffer() {
        SymmetricEncryptor enc = new SymmetricEncryptor();
        byte[] key = enc.generateKey();
        byte[] plaintext = "same message".getBytes(StandardCharsets.UTF_8);
        byte[] c1 = enc.encrypt(key, plaintext, null);
        byte[] c2 = enc.encrypt(key, plaintext, null);
        assertFalse(Arrays.equals(c1, c2));
    }
}
